include/swift/AST/Expr.h

```diff
@@ -1432,28 +1432,33 @@ class TypeExpr : public Expr {
 
 class TypeValueExpr : public Expr {
   TypeLoc paramTypeLoc;
+  Type valueType;
```

Comment:
I suppose the type of the expression is already the value type in some sense, but I added this because in CSGen and CSApply we already have the value type from the declaration that generated the TypeValueExpr so I used it as an assertion to ensure that the type we're setting this in CSApply is equal to the thing we've already found.